File: 2026/03- March/18-Wednesday/Solution.java
class Solution {
    public int countSubmatrices(int[][] grid, int k) {
        int m = grid.length;
        int n = grid[0].length;
        
        int[] colSum = new int[n];
        int count = 0;
        
        for (int i = 0; i < m; i++) {
            int rowPrefix = 0;
            
            for (int j = 0; j < n; j++) {
                // accumulate column sum
                colSum[j] += grid[i][j];
                
                // prefix sum from (0,0) to (i,j)
                rowPrefix += colSum[j];
                
                if (rowPrefix <= k) {
                    count++;
                }
            }
        }
        
        return count;
    }
}

File: 2026/03- March/18-Wednesday/solution.py
class Solution:
    def countSubmatrices(self, grid, k):
        m, n = len(grid), len(grid[0])
         
        col_sum = [0] * n
        count = 0
        
        for i in range(m):
            row_prefix = 0
            for j in range(n):
                # accumulate column sum
                col_sum[j] += grid[i][j]
                
                # prefix sum from (0,0) to (i,j)
                row_prefix += col_sum[j]
                
                if row_prefix <= k:
                    count += 1
        
        return count
